scenarios/kubernetes_kubelet.py

```diff
@@ -114,31 +113,86 @@ def add_gce_ssh(private_key, public_key):
     shutil.copy(public_key, gce_pub)
 
 
-def run_local_mode(script, properties, private_key, public_key):
+def parse_env(env):
+    """Returns (FOO, BAR=MORE) for FOO=BAR=MORE."""
+    return env.split('=', 1)
+
+
+def get_envs_from_file(env_file):
+    """Returns all FOO=BAR lines from env_file."""
+    envs = []
+    with open(env_file) as fp:
+        for line in fp:
+            line = line.rstrip()
+            if not line or line.startswith('#'):
+                continue
+            envs.append(parse_env(line))
+    return dict(envs)
+
+
+def set_defaults(envs):
+    '''Sets the default values for the environment variables if unset.'''
+    default_envs = {
+        'PARALLELISM': '8',
+        'WORKSPACE': '/tmp/',
+        'ARTIFACTS': '/tmp/_artifacts',
+        'TIMEOUT': '45m',
+    }
+    for name, value in default_envs.iteritems():
+        if name not in envs:
+            envs[name] = value
+
+
+def run_local_mode(env_file, ginkgo_flags, test_args, private_key, public_key):
     """Checkout, build and trigger the node e2e tests locally."""
     k8s = os.getcwd()
     if not os.path.basename(k8s) == 'kubernetes':
         raise ValueError(k8s)
     add_gce_ssh(private_key, public_key)
-    check('%s/%s' % (test_infra(), script), '%s/%s' % (test_infra(), properties))
+    envs = get_envs_from_file(os.path.join(test_infra(), env_file))
+    set_defaults(envs)
+    os.environ.update(envs)
+    check(
+        'go', 'run', 'test/e2e_node/runner/remote/run_remote.go',
+        '--logtostderr',
+        '--vmodule=*=4',
+        '--ssh-env=gce',
+        '--ssh-user=%s' % os.environ.get('GCE_USER', ''),
+        '--zone=%s' % os.environ.get('GCE_ZONE', ''),
+        '--project=%s' % os.environ.get('GCE_PROJECT', ''),
+        '--hosts=%s' % os.environ.get('GCE_HOSTS', ''),
+        '--images=%s' % os.environ.get('GCE_IMAGES', ''),
+        '--image-project=%s' % os.environ.get('GCE_IMAGE_PROJECT', ''),
+        '--image-config-file=%s' % os.environ.get('GCE_IMAGE_CONFIG_PATH', ''),
+        '--cleanup=%s' % os.environ.get('CLEANUP', ''),
+        '--results-dir=%s' % os.environ.get('ARTIFACTS', ''),
+        '--ginkgo-flags=--nodes=%s %s' % (
+            os.environ.get('PARALLELISM', ''),
+            ginkgo_flags),
+        '--test-timeout=%s' % os.environ.get('TIMEOUT', ''),
+        '--test_args=%s' % test_args,
+        '--instance-metadata=%s' % os.environ.get('GCE_INSTANCE_METADATA', ''),
+        '--system-spec-name=%s' % os.environ.get('SYSTEM_SPEC_NAME', ''),
```

Comment:
I wanted to make the parameters to as much consistent as possible between `kubernetes_e2e.py` and `kubernetes_kubelet.py`, so that I don't need to define two sections for each dimension (e.g., image, testSuites). So I thought we need to convert the flags between the scenario and the `go run` here.

But thinking more, maybe we can just live with this inconsistency. I will change the scenario to pass the unknown flags to `go run`.